react/src/ProgressIndicator/ProgressIndicator.tsx

```diff
@@ -0,0 +1,94 @@
+import { useMemo } from 'react'
+import { Box, BoxProps, useMultiStyleConfig } from '@chakra-ui/react'
+
+import { MotionBox } from '../motion'
+
+const ActiveIndicator = (): JSX.Element => {
+  const styles = useMultiStyleConfig('ProgressIndicator')
+  return <Box __css={styles.activeIndicator} />
+}
+
+interface CircleIndicatorProps extends BoxProps {
+  onClick: () => void
+  isActiveIndicator: boolean
+}
+
+const CircleIndicator = ({
+  onClick,
+  isActiveIndicator,
+  ...props
+}: CircleIndicatorProps): JSX.Element => {
+  const styles = useMultiStyleConfig('ProgressIndicator', { isActiveIndicator })
+  return (
+    <Box
+      __css={styles.circleIndicator}
+      onClick={onClick}
+      _hover={{ backgroundColor: 'secondary.300' }}
+      _focus={
+        isActiveIndicator
+          ? undefined
+          : {
+              backgroundColor: 'secondary.300',
+              boxShadow: `0 0 0 1px var(--chakra-colors-secondary-400)`,
+            }
+      }
+      as="button"
+      {...props}
+    />
+  )
+}
+
+export interface ProgressIndicatorProps {
+  /**
+   * The number of indicators to display
+   */
+  numIndicators: number
+  /**
+   * The current active indicator, this will be longer
+   */
+  currActiveIdx: number
+  /**
+   * Action to take when the indicator is clicked
+   * @param indicatorIdx The indicator that was clicked
+   */
+  onClick: (indicatorIdx: number) => void
+}
+
+export const ProgressIndicator = ({
+  numIndicators,
+  currActiveIdx,
+  onClick,
+}: ProgressIndicatorProps): JSX.Element => {
+  const styles = useMultiStyleConfig('ProgressIndicator')
+
+  const indicators = useMemo(
+    () => Array(numIndicators).fill(1),
+    [numIndicators],
+  )
+
+  const animationProps = useMemo(() => {
+    return { x: `${currActiveIdx + 0.125}rem` }
+  }, [currActiveIdx])
+
+  return (
+    <Box __css={styles.container}>
+      {indicators.map((_, idx) => (
+        <CircleIndicator
+          key={idx}
+          isActiveIndicator={idx === currActiveIdx}
```

Comment:
prefer using `_active` css pseudo prop instead of a property. To do this, we could set data-active instead:

```tsx
import { dataAttr } from '@chakra-ui/utils' // Required since any declaration of data-* is considered truthy, and this util removes the prop on any falsey value

...

<CircleIndicator
  ...
  data-active={dataAttr(idx === currActiveIdx)}
/>
```

Reply:
thanks for this! i think i need to be more cognizant about using pseudo states